cadence-contracts/PackNFT.cdc

```diff
@@ -129,13 +130,110 @@ pub contract PackNFT: NonFungibleToken, IPackNFT {
             self.issuer = issuer
         }
 
+
+        // All supported metadata views for the Moment including the Core NFT Views
+        //
+        pub fun getViews(): [Type] {
+            return [
+                Type<MetadataViews.Display>(),
+                Type<MetadataViews.ExternalURL>(),
+                Type<MetadataViews.Medias>(),
+                Type<MetadataViews.NFTCollectionData>(),
+                Type<MetadataViews.NFTCollectionDisplay>(),
+                Type<MetadataViews.Royalties>(),
+                Type<MetadataViews.Serial>(),
+            ]
+        }
+
+        pub fun resolveView(_ view: Type): AnyStruct? {
+            switch view {
+                case Type<MetadataViews.Display>():
+                    return MetadataViews.Display(
+                        name: "NFL All Day Pack",
+                        description: "Reveals official NFL All Day Moments when opened",
+                        thumbnail: MetadataViews.HTTPFile(url: self.getImage(imageType: "image", format: "jpeg", width: 256))
+                    )
+                case Type<MetadataViews.ExternalURL>():
+                    return MetadataViews.ExternalURL("https://nflallday.com/packNFTs/".concat(self.id.toString())) // might have to make a URL that redirects to packs page based on packNFT id -> distribution id
```

Comment:
need to implement this page that takes a packNFT's flow id and directs a user to the pack's distribution's page